src/Adapter/Currency/CommandHandler/EditCurrencyHandler.php

```diff
@@ -0,0 +1,233 @@
+                $entity->iso_code,
+                $command->getIsoCode()->getValue()
+            );
+            $this->assertDefaultCurrencyIsNotBeingDisabled($command->getCurrencyId()->getValue(), $command->isEnabled());
+            $this->assertDefaultCurrencyIsBeingRemovedFromShop(
+                $entity,
+                $command->getShopIds()
+            );
+
+            $entity->iso_code = $command->getIsoCode()->getValue();
+            $entity->active = $command->isEnabled();
+            $entity->conversion_rate = $command->getExchangeRate()->getValue();
+
+            if (false === $entity->update()) {
+                throw new CannotUpdateCurrencyException(
+                    sprintf(
+                        'An error occurred when updating currency object with id "%s"',
+                        $command->getCurrencyId()->getValue()
+                    )
+                );
+            }
+
+            $this->associateWithShops($entity, $command->getShopIds());
+            $this->associateConversionRateToShops($entity, $command->getShopIds());
+        } catch (PrestaShopException $exception) {
+            throw new CurrencyException(
+                sprintf(
+                    'An error occurred when updating currency object with id "%s"',
+                    $command->getCurrencyId()->getValue()
+                ),
+                0,
+                $exception
+            );
+        }
+
+        return new CurrencyId((int) $entity->id);
+    }
+
+    /**
+     * When editing the currency it checks if new iso code does not exist
+     * so it will not create multiple currencies with same iso codes,
+     *
+     * @param int $currencyId
+     * @param string $currentIsoCode
+     * @param string $newIsoCode
+     *
+     * @throws CurrencyConstraintException
+     */
+    private function assertCurrencyWithIsoCodeDoesNotExist($currencyId, $currentIsoCode, $newIsoCode)
+    {
+        if ($currentIsoCode === $newIsoCode) {
+            return;
+        }
+
+        $qb = new DbQuery();
+        $qb
+            ->select('id_currency')
+            ->from('currency')
+            ->where('id_currency !=' . $currencyId)
+            ->where('iso_code = "' . pSQL($newIsoCode) . '"')
+        ;
+
+        $result = Db::getInstance()->getValue($qb);
+
+        if (is_numeric($result)) {
+            throw new CurrencyConstraintException(
+                sprintf(
+                    'Currency with iso code "%s" already exist and cannot be created',
+                    $newIsoCode
+                ),
+                CurrencyConstraintException::CURRENCY_ALREADY_EXISTS
+            );
+        }
+    }
+
+    /**
+     * Prevents from default currency being disabled.
+     *
+     * @param int $currencyId
+     * @param bool $isEnabled
+     *
+     * @throws CannotDisableDefaultCurrencyException
+     */
+    private function assertDefaultCurrencyIsNotBeingDisabled($currencyId, $isEnabled)
```

Comment:
i think you can pass whole command instead of 2 args since you want to assert that command is not disabling default currency, wdyt?